### PR TITLE
Add Escape Character feature

### PR DESCRIPTION
**Pull request by Danieladu**
Currently we have a requirement to allow users to use escape characters in .lg files like `\r`, `\n`,`\t` and so on.

The current rule is like this：`\r`,`\n`,`\t` we will keep its original style.
the other format starts with `\` ,The second character will be parsed. `\\`->`\`，`\[`->`[`

### CODE REVIEW

File: libraries/Microsoft.Bot.Builder.AI.LanguageGeneration/Evaluator.cs [hunk 1]
```diff
@@ -128,13 +131,33 @@ public override string VisitNormalTemplateString([NotNull] LGFileParser.NormalTe
                         builder.Append(EvalMultiLineText(node.GetText()));
                         break;
                     default:
-                        builder.Append(node.GetText());
+                        builder.Append(EvalEscapeCharacter(node.GetText()));
```
Comment (boydc2014) [inline libraries/Microsoft.Bot.Builder.AI.LanguageGeneration/Evaluator.cs]: Why do you need to change the default behavior of evalution?

File: libraries/Microsoft.Bot.Builder.AI.LanguageGeneration/Evaluator.cs [hunk 2]
```diff
@@ -128,13 +131,33 @@ public override string VisitNormalTemplateString([NotNull] LGFileParser.NormalTe
...
                         break;
                 }
             }
             return builder.ToString();
         }
-        
+
+        private string EvalEscapeCharacter(string exp) => Regex.Replace(exp, @"\\[\w\\\[\{\]}]", new MatchEvaluator(EscapeMatcher));
```
Comment (boydc2014) [inline libraries/Microsoft.Bot.Builder.AI.LanguageGeneration/Evaluator.cs]: do you still need \\ even with @ ?

File: libraries/Microsoft.Bot.Builder.AI.LanguageGeneration/Evaluator.cs [hunk 3]
```diff
@@ -128,13 +131,33 @@ public override string VisitNormalTemplateString([NotNull] LGFileParser.NormalTe
...
+
+
+        private string EscapeMatcher(Match match)
+        {
+            var noEscape = new List<string> { "\\r", "\\n", "\\t" };
```
Comment (boydc2014) [inline libraries/Microsoft.Bot.Builder.AI.LanguageGeneration/Evaluator.cs]: use @ to simplify that?

File: libraries/Microsoft.Bot.Builder.AI.LanguageGeneration/Evaluator.cs (hunk 1 above)
Comment (Danieladu) [reply libraries/Microsoft.Bot.Builder.AI.LanguageGeneration/Evaluator.cs]: I have deleted it at the newest version

**Pull request merged**